File: Source/Extensions/Tessa.Extensions.Default.Client/UI/OutgoingPartnerUIExtension.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using Tessa.Cards;
using Tessa.UI.Cards;

namespace Tessa.Extensions.Default.Client.UI
{
    public class OutgoingPartnerUIExtension : CardUIExtension
    {
        #region Base Overrides

        public override async Task Initializing(ICardUIExtensionContext context)
        {
            var dciSection = context.Card.Sections["DocumentCommonInfo"];
            if (dciSection == null)
            {
                return;
            }

            dciSection.FieldChanged += (s, e) =>
            {
                if (e.FieldName == "PartnerID")
                {
                    IDictionary<string, object> fields = ((CardSection)s).Fields;
                    fields["ReceiverName"] = null;
                    fields["ReceiverRowID"] = null;
                }
            };
        }

        #endregion
    }
}

File: WebClient SDK/src/default/ui/outgoingPartnerUIExtension.ts
import { CardUIExtension, ICardUIExtensionContext } from 'tessa/ui/cards';

export class OutgoingPartnerUIExtension extends CardUIExtension {

  public initialized(context: ICardUIExtensionContext) {
    if (context.card.typeId !== 'c59b76d9-c0db-01cd-a3fb-b339740f0620') { // OutgoingTypeID
      return;
    }

    const dciSection = context.card.sections.tryGet('DocumentCommonInfo');
    if (!dciSection) {
      return;
    }

    dciSection.fields.fieldChanged.add(e => {
      if (e.fieldName === 'PartnerID') {
        e.storage.set('ReceiverName', null);
        e.storage.set('ReceiverRowID', null);
      }
    });
  }

}